src/bin/cargo-xrustc.rs

```diff
@@ -0,0 +1,171 @@
+extern crate cargo_metadata;
+#[macro_use]
+extern crate error_chain;
+extern crate fs2;
+#[cfg(any(all(target_os = "linux", not(target_env = "musl")), target_os = "macos"))]
+extern crate libc;
+extern crate rustc_version;
+extern crate serde_json;
+#[macro_use]
+extern crate serde_derive;
+extern crate tempdir;
+extern crate toml;
+extern crate walkdir;
+extern crate xargo_lib;
+
+use std::hash::{Hash, Hasher};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::ExitStatus;
+use std::{env, io, process};
+
+use rustc_version::Channel;
+
+use xargo_lib::errors::*;
+use xargo_lib::rustc::Target;
+
+use xargo_lib::cargo;
+use xargo_lib::cli;
+use xargo_lib::config;
+use xargo_lib::errors;
+use xargo_lib::extensions;
+use xargo_lib::flock;
+use xargo_lib::rustc;
+use xargo_lib::sysroot;
+use xargo_lib::util;
+use xargo_lib::xargo;
+
+use xargo_lib::CurrentDirectory;
+use xargo_lib::CompilationMode;
+
+const HELP: &str = include_str!("help.txt");
+
+
+
+pub fn main() {
```

Comment:
I think most of the `main`, `run`, and `build` methods could be moved to the lib (same for the `cargo-build` binary`).